netbox_onboarding/worker.py

```diff
@@ -29,43 +30,58 @@ def onboard_device(task_id, credentials):
     """Process a single OnboardingTask instance."""
     username = credentials.username
     password = credentials.password
+    secret = credentials.secret
 
-    try:
-        ot = OnboardingTask.objects.get(id=task_id)
-    except OnboardingTask.DoesNotExist:
-        # TODO: maybe we started before the DB was done writing it, or maybe it was deleted out from under us?
-        time.sleep(1)
-        ot = OnboardingTask.objects.get(id=task_id)
+    ot = OnboardingTask.objects.get(id=task_id)
 
-    logging.info("START: onboard device")
+    logger.info("START: onboard device")
+    onboarded_device = None
+    try:
+        if ot.ip_address:
+            onboarded_device = Device.objects.get(
+                primary_ip4__address__net_host=ot.ip_address
+            )
+    except:
```

Comment:
Is there an exception other than `Device.DoesNotExist` that we would expect and need to catch here?

Reply:
At least a few: DoesNotExist, multiple objects returned, ValueError and ValidationError for non-ipaddresses (in model IPAddr is CharField)